### PR TITLE
feat: Add serialization traits to Status

### PR DESCRIPTION
**Pull request by bitdivine**
# Description

We would like to express the Status response printed by dfx as IDL and/or JSON.  This requires corresponding serialization traits to be implemented for Status.

Contributes towards fixing: https://dfinity.atlassian.net/browse/SDK-1319

# How Has This Been Tested?

Unit tests are included for serialization to JSON and IDL.

# Checklist:

- [x] The title of this PR complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/).
- [x] I have edited the CHANGELOG accordingly.
- [x] I have made corresponding changes to the documentation.
  - Not sure what is needed here, if anything.  The traits should show up in the generated rustdoc and that is the documentation I look at most frequently when I wish to know how to serialize a type.  On that basis I believe this is done.

### CODE REVIEW

File: ic-agent/src/agent/status.rs [hunk 1]
```diff
@@ -55,6 +59,37 @@ pub struct Status {
...
+        replica_health_status: None,
+        root_key: None,
+        values: BTreeMap::new(),
+    };
+    let expected_json =
+        r#"{"impl_version":null,"replica_health_status":null,"root_key":null,"values":{}}"#;
+    let actual_json = serde_json::to_string(&status).expect("Failed to serialize as JSON");
+    assert_eq!(expected_json, actual_json);
+}
+#[test]
```
[the remark this answers is not in the document]
Comment (bitdivine) [reply ic-agent/src/agent/status.rs:75]: Ok, found out how to convert.  Surprisingly difficult and inefficient.

[the remark this answers is not in the document]
Comment (bitdivine) [reply ic-agent/src/agent/status.rs:75]: Added the conversion method to the Candid library, with improvements by YanChen.  Efficiency improvements can be applied there.

File: ic-agent/Cargo.toml [hunk 2]
```diff
@@ -17,7 +17,7 @@ include = ["src", "Cargo.toml", "../LICENSE", "README.md"]
 [dependencies]
 backoff = "0.4.0"
 cached = { version = "0.46", features = ["ahash"], default-features = false }
-candid = { workspace = true }
+candid = { workspace = true, features = ["value"] }
```
[the remark this answers is not in the document]
Comment (bitdivine) [reply ic-agent/Cargo.toml:20]: Yes:  Experiment:
 -  I dropped the "values" feature from the candid dependency.
 - I added a the candid crate under an alias in the dev dependencies.
 - cargo check complained about the duplicate:
```
max@sinkpad:~/dfn/agent-rs (14:27)$ git diff
diff --git a/ic-agent/Cargo.toml b/ic-agent/Cargo.toml
index 8308d2b..5b41a89 100644
--- a/ic-agent/Cargo.toml
+++ b/ic-agent/Cargo.toml
@@ -17,7 +17,7 @@ include = ["src", "Cargo.toml", "../LICENSE", "README.md"]
 [dependencies]
 backoff = "0.4.0"
 cached = { version = "0.46", features = ["ahash"], default-features = false }
-candid = { workspace = true, features = ["value"] }
+candid = { workspace = true }
 ed25519-consensus = { version = "2" }
 futures-util = "0.3.21"
 hex = { workspace = true }
@@ -75,6 +75,7 @@ web-sys = { version = "0.3", features = ["Window"], optional = true }
 
 [dev-dependencies]
 serde_json = "1.0.79"
+dev_candid = { package = "candid", version = "0.10.0", features = ["value"]}
 
 [target.'cfg(not(target_family = "wasm"))'.dev-dependencies]
 tokio = { version = "1.24.2", features = ["full"] }
diff --git a/ic-agent/src/agent/status.rs b/ic-agent/src/agent/status.rs
index e54ea25..a53d2f3 100644
--- a/ic-agent/src/agent/status.rs
+++ b/ic-agent/src/agent/status.rs
@@ -74,8 +74,8 @@ fn can_serilaize_status_as_json() {
 }
 #[test]
 fn can_serialize_status_as_idl() {
-    use candid::types::value::IDLValue;
-    use candid::{Encode, IDLArgs, Result as CandidResult, TypeEnv};
+    use dev_candid::types::value::IDLValue;
+    use dev_candid::{Encode, IDLArgs, Result as CandidResult, TypeEnv};
     let status = Status {
         impl_version: Some("Foo".to_string()),
         replica_health_status: None,
max@sinkpad:~/dfn/agent-rs (14:29)$ 
max@sinkpad:~/dfn/agent-rs (14:32)$ 
max@sinkpad:~/dfn/agent-rs (14:32)$ cargo check
error: the crate `ic-agent v0.31.0 (/home/max/dfn/agent-rs/ic-agent)` depends on crate `candid v0.10.0` multiple times with different names
max@sinkpad:~/dfn/agent-rs (14:38)$ 
```

[the remark this answers is not in the document]
Comment (smallstepman) [reply ic-agent/Cargo.toml:20]: this worked for me
image

[the remark this answers is not in the document]
Comment (smallstepman) [reply ic-agent/Cargo.toml:20]: btw, I don't have strong opinion about adding `value` feature despite not being used in production code... maybe its ok and not worth the hassle to play the cargo here. Was only trying to be helpful ;)

**Review by ericswanson-dfinity — APPROVED** [review]

**Pull request merged**